File: fixtures/native-event.swift
struct Progress {
  var percent: Double
  var label: String
}

func __event_253846265b0a3b85_progress(payload: Progress) throws -> Void {
  try LucentEventHub.shared.emit("__event_253846265b0a3b85_progress", ["percent": payload.percent, "label": payload.label] as [String: Any])
}

func report(percent: Double) throws -> Void {
  _ = try __event_253846265b0a3b85_progress(payload: Progress(percent: percent, label: "native"))
}

File: fixtures/native-event.kt
data class Progress(
  var percent: Double,
  var label: String
)

fun __event_253846265b0a3b85_progress(payload: Progress): Unit {
  LucentEventHub.emit("__event_253846265b0a3b85_progress", mapOf("percent" to payload.percent, "label" to payload.label))
}

fun report(percent: Double): Unit {
  __event_253846265b0a3b85_progress(Progress(percent = percent, label = "native"))
}
